x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -606,19 +611,12 @@ void handleNullToken() {
                 authentication = null;
             }
 
-            Runnable action;
             if (authentication != null) {
-                action = () -> writeAuthToContext(authentication);
+                writeAuthToContext(authentication);
             } else {
-                action = () -> {
-                    logger.debug("No valid credentials found in request [{}], rejecting", request);
-                    listener.onFailure(request.anonymousAccessDenied());
-                };
+                logger.debug("No valid credentials found in request [{}], rejecting", request);
+                listener.onFailure(request.anonymousAccessDenied());
             }
-
-            // we assign the listener call to an action to avoid calling the listener within a try block and auditing the wrong thing when
-            // an exception bubbles up even after successful authentication
```

Comment:
Don't quite get this existing comment. But I suspect the original concern may not exist any more.